bofire/data_models/strategies/stepwise/conditions.py

```diff
@@ -1,9 +1,18 @@
-from typing import List, Literal, Union
+from abc import abstractmethod
+from typing import List, Literal, Optional, Union
 
+import pandas as pd
 from pydantic import Field, field_validator
 from typing_extensions import Annotated
 
 from bofire.data_models.base import BaseModel
+from bofire.data_models.domain.api import Domain
```

Comment:
Why do you have this extra class and not just put the abstract method in `Condition?`

Reply:
I prefer abstract classes to be really abstract and to be mixed with data members or implementations.